packages/legend-application-data-cube/src/stores/LegendDataCubeDataCubeEngine.ts

```diff
@@ -379,6 +389,24 @@ export class LegendDataCubeDataCubeEngine extends DataCubeEngine {
         source.parameterValues,
         options,
       );
+    } else if (source instanceof CachedDataCubeSource) {
+      //execute plan
+      const executionPlan = await this.generateExecutionPlan(
+        query,
+        source.model,
+        [],
+        options,
+      );
+      const sql = this.getSqlFromExecutionPlan(executionPlan);
+      if (sql !== undefined) {
+        return {
+          executedQuery: await queryCodePromise,
+          executedSQL: sql,
+          result: await this._cacheManager.runQuery(sql),
+        };
+      } else {
+        throw new UnsupportedOperationError(`Can't execute generated sql`);
```

Comment:
Don't think this is the right error message, also idiomaticly, let's do assertion to be clean.

```ts
assertNonNullable(sql, `Can't extract generated SQL`)
```

Also, there's little value in creating a method to get the SQL, just inline the logic here.